mobile/src/components/Button.tsx

```diff
@@ -1,31 +1,33 @@
 import { ReactNode } from "react";
-import { GestureResponderEvent, Pressable, Text } from "react-native";
+import { Pressable, Text, PressableProps } from "react-native";
 
-interface Props {
+interface Props extends PressableProps {
   name: string;
-  className: string;
-  children: ReactNode[];
-  onPress: (event: GestureResponderEvent) => void;
+  className?: string;
+  children?: ReactNode;
   disabled?: boolean;
   pilled?: boolean;
 }
 
 /**
- * buttonName - name of the button.
- * addButtonStyle - add custom style to the button.
- * children - if you want to add your own text with styles.
- * pilled - make the button pilled (optional).
- * disabled - greys out button (optional).
+ * name - (string) button text (required).
+ * className - (string) additional button styling (required).
+ * children - (jsx tags/component)(optional) ie. pass in an icon (optional).
+ * onPress - (function) button click handler. (optional) (default: ()=>{})
+ * disabled - (boolean) calling this in the prop will grey the button out(optional).
+ * pilled - (boolean) calling this will round the button all the way(optional).
  */
+
 export default function Button({
   name,
-  className,
-  children,
-  onPress,
+  className = "",
+  children = null,
   disabled = false,
   pilled = false,
+  ...props
 }: Props) {
-  let baseStyle = "h-7 items-center justify-center";
+  //warning: do not touch the extra spaces at the end or it will break the styling
+  let baseStyle = "h-7 items-center justify-center ";
```

Comment:
I recommend you add the spacing in front wherever you append more styles instead of at the end like this.